declare that OAM runtime could have more extended application scopes (#263)
oam-dev/spec: 4.application_scopes.md

```

## Extended application scope type definitions

The following extended scope types are available:
- resource quota scope
- identity scope

> These two are just definition examples, the runtime could have more extended application scopes as they need.

Reviewer: ```suggestion
Besides the above scopes, the runtime could have more extended application scopes as needed.
```
